src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/TraceLocationDetailViewController.swift

```diff
@@ -0,0 +1,183 @@
+	) {
+		self.dismiss = dismiss
+		self.presentCheckins = presentCheckins
+		self.viewModel = TraceLocationDetailViewModel(traceLocation)
+		
+		super.init(nibName: nil, bundle: nil)
+	}
+
+	@available(*, unavailable)
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+	
+	// MARK: - Overrides
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		setupView()
+	}
+	
+	// MARK: - Private
+	
+	@IBOutlet private weak var bottomCardView: UIView!
+	@IBOutlet private weak var descriptionView: UIView!
+	@IBOutlet private weak var logoImageView: UIImageView!
+	@IBOutlet private weak var checkInForLabel: ENALabel!
+	@IBOutlet private weak var activityLabel: ENALabel!
+	@IBOutlet private weak var descriptionLabel: ENALabel!
+	@IBOutlet private weak var addressLabel: ENALabel!
+	@IBOutlet private weak var saveToDiaryLabel: ENALabel!
+	@IBOutlet private weak var automaticCheckOutLabel: ENALabel!
+	@IBOutlet private weak var pickerButton: ENAButton!
+	@IBOutlet private weak var pickerContainerView: UIView!
+	@IBOutlet private weak var pickerSeparator: UIView!
+	@IBOutlet private weak var datePickerView: UIDatePicker!
+	@IBOutlet private weak var additionalInfoView: UIView!
+	@IBOutlet private weak var additionalInfoLabel: ENALabel!
+	@IBOutlet private weak var pickerSwitch: ENASwitch!
+		
+	private let viewModel: TraceLocationDetailViewModel
+	private let dismiss: () -> Void
+	private let presentCheckins: () -> Void
+	private var subscriptions = Set<AnyCancellable>()
+	private var selectedDuration: Int?
+	private var isInitialSetup = true
+
+	private func setupView() {
+		viewModel.setupView()
+
+		view.backgroundColor = .enaColor(for: .background)
+		checkInForLabel.text = AppStrings.Checkins.Details.checkinFor
+		activityLabel.text = AppStrings.Checkins.Details.activity
+		saveToDiaryLabel.text = AppStrings.Checkins.Details.saveToDiary
+		automaticCheckOutLabel.text = AppStrings.Checkins.Details.automaticCheckout
+		pickerButton.setTitleColor(.enaColor(for: .textPrimary1), for: .normal)
+		logoImageView.image = logoImageView.image?.withRenderingMode(.alwaysTemplate)
+		logoImageView.tintColor = .enaColor(for: .textContrast)
+		addBorderAndColorToView(descriptionView, color: .enaColor(for: .hairline))
+		addBorderAndColorToView(bottomCardView, color: .enaColor(for: .hairline))
+		addBorderAndColorToView(additionalInfoView, color: .enaColor(for: .hairline))
+
+		setupAdditionalInfoView()
+		
+		viewModel.$descriptionLabelTitle
+			.sink { [weak self] description in
+				self?.descriptionLabel.text = description
+			}
+			.store(in: &subscriptions)
+		
+		viewModel.$addressLabelTitle
+			.sink { [weak self] address in
+				self?.addressLabel.text = address
+			}
+			.store(in: &subscriptions)
+		
+		viewModel.$initialDuration
+			.sink { [weak self] duration in
+				self?.selectedDuration = duration
+				self?.setupPicker(with: duration ?? 0)
+
+			}
+			.store(in: &subscriptions)
```

Comment:
Are these ever updated? If not, you could just assign them once without using combine.

Reply:
I wanted to keep the TraceLocation object Private and only accessible from the viewModel, but ok i can add some properties in the class from the init method